integration-tests/src/test/java/org/apache/fineract/integrationtests/inlinecob/InlineLoanCOBTest.java

```diff
@@ -185,7 +185,7 @@ public void testInlineCOBCatchUpLoans() {
             final GetDelinquencyBucketsResponse delinquencyBucket = DelinquencyBucketsHelper.getDelinquencyBucket(requestSpec, responseSpec,
                     delinquencyBucketResponse.getResourceId());
 
-            final Integer loanProductID = createLoanProduct(loanTransactionHelper, delinquencyBucket.getId());
+            final Integer loanProductID = createLoanProduct(loanTransactionHelper, Math.toIntExact(delinquencyBucket.getId()));
```

Comment:
We may also update the LoanProductTestBuilder to use Longs for the bucket identifiers.

Reply:
As agreed it will be separated into an other task